src/Messages/Interactions/TSMessage.m

```diff
@@ -131,7 +132,17 @@ - (nullable instancetype)initWithCoder:(NSCoder *)coder
         _attachmentIds = [NSMutableArray new];
     }
 
+    if (!_receivedAtDate) {
+        // TSIncomingMessage.receivedAt has been superceded by TSMessage.receivedAtDate.
+        NSDate *receivedAt = [coder decodeObjectForKey:@"receivedAt"];
+        _receivedAtDate = receivedAt;
+    }
+
     _schemaVersion = OWSMessageSchemaVersion;
+
+    // We _DO NOT_ set _receivedAt_ in this constructor.  We don't want to
+    // set the receivedAt time for old messages in the data store.
```

Comment:
This comment can probably go away?

OK. seriously though, last nit from me. Thanks for your patience, I'm really happy with where this ended up.